SantasBag/SantasBag.WebHost/Controllers/RewardsController.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using SantasBag.Contracts;
+using SantasBag.Core.Models;
+using SantasBag.WebHost.Contracts;
+using SantasBug.Core.Abstractions;
+
+namespace SantasBag.Controllers;
+
+[ApiController]
+[Route("controller")]
+public class RewardsController : ControllerBase
+{
+    private readonly IRewardsService _rewardsService;
+    public RewardsController(IRewardsService rewardsService)
+    {
+        _rewardsService = rewardsService;
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<List<RewardResponse>>> GetRewards(CancellationToken cancellationToken)
+    {
+        var rewards = await _rewardsService.GetAllRewards(cancellationToken);
```

Comment:
I wonder if the token will come here or if it still needs to be taken from HttpContext?

Reply:
> I wonder if the token will come here or if it still needs to be taken from HttpContext?

How i understand, yes. And i guess there is no need to transfer the cancellation token to the method